app/helpers/branded-preprint-url.js

```diff
@@ -0,0 +1,25 @@
+import Ember from 'ember';
+import pathJoin from '../utils/path-join';
+
+/**
+ * @module ember-preprints
+ * @submodule helpers
+ */
+
+/**
+ * To determine the link url for branded preprints on the provider carousel
+ *
+ * @class route-prefix
+ */
+
+export function brandedPreprintUrl(provider, lightLogo) {
+    let url = '';
+    if (provider[0].get('domain') && provider[0].get('domainRedirectEnabled')) {
+        url = lightLogo.type ? provider[0].get('domain') : pathJoin(provider[0].get('domain'), 'discover');
+    } else {
+        url = lightLogo.type ? '/preprints/' + provider[0].get('id') : '/preprints/' + provider[0].get('id') + '/discover';
+    }
```

Comment:
I think the logic would be cleaner is you first build the base url and then append `/discover` if necessary.